cmake/CMakeLists.txt

```diff
@@ -425,9 +425,11 @@ if (WIN32)
 
     # set linker flags to minimize the binary size.
     if (MSVC)
-      foreach(type EXE SHARED)
-        set(CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO} /OPT:REF,ICF,LBR")
-        set(CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_${type}_LINKER_FLAGS_RELWITHDEBINFO} /INCREMENTAL:NO")
+      foreach(type EXE STATIC SHARED)
+        if (NOT type MATCHES STATIC)
```

Comment:
>STATIC [](start = 29, length = 6)

does this affect other binaries or just ours?

Reply:
adding the STATIC in here makes it affect all static link libs.    it seemed to make sense to apply everywhere.    the problem is the linker has to take 2 passes if it needs to use LTCG and you didn't tell it before hand.  so all the static link libs where starting their link phase, then resetting once they realized they needed LTCG.   

---
In reply to: [347038839](https://github.com/microsoft/onnxruntime/pull/2406#discussion_r347038839) [](ancestors = 347038839)